cmd/assaf-http/main.go

```diff
@@ -392,9 +346,12 @@ type CleanQuery struct {
 	values  url.Values
 }
 
-func requestHandler(consumeRequests chan ConsumeRequest, msgQueue chan<- message) http.Handler {
+func requestHandler(msgQueue chan<- models.Message, tracer opentracing.Tracer, requests *Reqs) http.Handler {
+	rand.Seed(time.Now().Unix())
```

Comment:
another subtle bug, I was calling the seed functions on each _request_ while ugly doesn't sound too bad but considering the concurrent nature of the applications sometimes requests would come at the exact same time meaning the have the same random seed and the same id which most of the infinite loops above (but not all) since there were two requests with the same id the first of the requests would be returned leaving the other request with the same id looping for ever (unless it was dumped but it's pretty hard to check!)